src/Corax/Queries/SortingMultiMatch.cs

```diff
@@ -294,7 +296,7 @@ public int Compare(Item ix, Item iy)
             var sorter = new Sorter<MultiMatchComparer<TComparer1, W>.Item, long, MultiMatchComparer<TComparer1, W>>(comparer);
             sorter.Sort(matchesKeys[0..totalMatches], matches);
 
-            Span<long> bValues = stackalloc long[matches.Length];
+            Span<long> bValues = MemoryMarshal.Cast<byte, long>(bufferHolder.AsSpan().Slice(itemArraySize, matches.Length));
```

Comment:
Should be: `sizeof(long) * matches.Length`

Reply:
That's correct

Reply:
@maciejaszyk Nice catch!

Reply:
Guys, any idea _why_ none of tests has failed ?!?! I would definitely expect to see some error or memory access issues with that kind of bug

![image](https://user-images.githubusercontent.com/376551/146146479-1cdab1f8-6ed2-45d7-93d9-8c124586495d.png)

Reply:
More complex tests are done in https://github.com/ravendb/ravendb/pull/13207

Reply:
Unless we use them with straight pointer arithmetic's (like in this case) they wont fail but would make the algorithm requiring more steps... so I added a few more checks here and there.